### DISCUSSION TITLE
Running shell scripts in plugins file for particular spec on cypress run and cypress open

### DISCUSSION THREAD

**Discussion by shubham-lagshetti-zz** [cypress-io/cypress#21580] [Questions and Help]
**questions- Can we write plugins in such way that they can run for only particular spec file not just for all tests on cypress run and cypress open commands?**      

-note : i am new to the cypress still learning things.
- Problem:

In my project i am trying to do some event creation and deletion and validation after a particular spec file execution.
current what i did i write the shell script in plugins/index.js file as below"
`module.export=(on,config)=>{
on('before:spec',(spec)=>{
---------------------
//shell scripts 
-------------------
}

on('after:spec',(spec)=>{
------------------
//shell script
----------------
}
}
`

**this is what i am currently did but the shell script is running for every spec file on npx cypress run command i dont want that i want it to run for specific spec file event on execution of test via cypress runnner 

looking for solution.**

**Comment by muratkeremozcan** [comment]
_specific spec file event on execution of test_ : not sure what this means

You want per spec file script execution capability? cy.exec might be what you need https://docs.cypress.io/api/commands/exec.

**Reply by shubham-lagshetti-zz** [reply]
thanks for suggetion @muratkeremozcan   i used cy.exec command also but the error i am getting is as below screen shot 

that run:azure script will run an sample.sh file in package.json i added the scripts like this in package.json file 
 `scripts{
"run:azure":"sh "sample.sh file path " 
}
`
If there is any other way that we  can execute the shell script in spec files let me know that help me 
![image](https://user-images.githubusercontent.com/100254384/170528318-2e30b9ab-895d-449a-b410-53fee4a294f4.png)
